### PR TITLE
[php8-compact] Add in guards into various templates to fix issues in …

### PR DESCRIPTION
**Pull request by seamuslee001**
…test runs on php8

Overview
----------------------------------------
This adds some more if guards into various templates to support test runs in php8

ping @eileenmcnaughton @totten @colemanw @demeritcowboy

### CODE REVIEW

**Comment by seamuslee001** [comment]
jenkins test this please

File: templates/CRM/Group/Form/Edit.tpl [hunk 1]
```diff
@@ -22,7 +22,7 @@
     <tr class="crm-group-form-block-title">
       <td class="label">{$form.title.label} {if $action == 2}{include file='CRM/Core/I18n/Dialog.tpl' table='civicrm_group' field='title' id=$group.id}{/if}</td>
       <td>{$form.title.html|crmAddClass:huge}
-        {if $group.saved_search_id}&nbsp;({ts}Smart Group{/ts}){/if}
+        {if !empty($group.saved_search_id)}&nbsp;({ts}Smart Group{/ts}){/if}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Edit.tpl:25]: ok

File: templates/CRM/Report/Page/Options.tpl [hunk 2]
```diff
@@ -39,12 +39,12 @@
...
         </thead>
         {foreach from=$rows item=row}
-          <tr id="option_value-{$row.id}" class="crm-entity {cycle values="odd-row,even-row"}{$row.class}{if NOT $row.is_active} crm-report-optionList crm-report-optionList-status_disable disabled{else} crm-report-optionList crm-report-optionList-status_enable{/if}">
+          <tr id="option_value-{$row.id}" class="crm-entity {cycle values="odd-row,even-row"}{if !empty($row.class)}{$row.class}{/if}{if NOT $row.is_active} crm-report-optionList crm-report-optionList-status_disable disabled{else} crm-report-optionList crm-report-optionList-status_enable{/if}">
             <td class="crm-report-optionList-label crm-editable" data-field="label">{$row.label}</td>
             <td class="crm-report-optionList-value">{$row.value}</td>
             <td class="crm-report-optionList-description">{$row.description}</td>
             <td class="nowrap crm-report-optionList-order">{$row.weight}</td>
-            {if $showIsDefault}
+            {if !empty($showIsDefault)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Report/Page/Options.tpl:47]: ok

File: templates/CRM/Report/Page/Options.tpl [hunk 3]
```diff
@@ -39,12 +39,12 @@
         </tr>
         </thead>
         {foreach from=$rows item=row}
-          <tr id="option_value-{$row.id}" class="crm-entity {cycle values="odd-row,even-row"}{$row.class}{if NOT $row.is_active} crm-report-optionList crm-report-optionList-status_disable disabled{else} crm-report-optionList crm-report-optionList-status_enable{/if}">
+          <tr id="option_value-{$row.id}" class="crm-entity {cycle values="odd-row,even-row"}{if !empty($row.class)}{$row.class}{/if}{if NOT $row.is_active} crm-report-optionList crm-report-optionList-status_disable disabled{else} crm-report-optionList crm-report-optionList-status_enable{/if}">
```
Comment (eileenmcnaughton) [inline templates/CRM/Report/Page/Options.tpl:42]: ok

File: templates/CRM/Report/Page/Options.tpl [hunk 4]
```diff
@@ -28,7 +28,7 @@
           <th>{ts}URL{/ts}</th>
           <th>{ts}Description{/ts}</th>
           <th>{ts}Order{/ts}</th>
-          {if $showIsDefault}
+          {if !empty($showIsDefault)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Report/Page/Options.tpl:31]: ok

File: templates/CRM/Pledge/Page/UserDashboard.tpl [hunk 5]
```diff
@@ -52,7 +52,7 @@
 {*pledge row if*}
 
 {*Display honor block*}
-{if $pledgeHonor && $pledgeHonorRows}
+{if !empty($pledgeHonor) && !empty($pledgeHonorRows)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Pledge/Page/UserDashboard.tpl:55]: ok

File: templates/CRM/Member/Page/MembershipStatus.tpl [hunk 6]
```diff
@@ -38,17 +38,17 @@
...
+          <td class="nowrap crmf-start_event_adjust_unit_interval">{if !empty($start_event_adjust_unit_interval)}{$row.start_event_adjust_unit_interval}{/if}</td>
+          <td class="nowrap crmf-end_event crm-editable" data-type="select" data-empty-option="{ts}- none -{/ts}">{if !empty($row.end_event)}{$row.end_event}{/if}</td>
+          <td class="nowrap crmf-end_event_adjust_interval">{if !empty($end_event_adjust_unit_interval)}{$row.end_event_adjust_interval}{/if}</td>
           <td class="crmf-is_current_member crm-editable" data-type="boolean">{if $row.is_current_member eq 1} {ts}Yes{/ts} {else} {ts}No{/ts} {/if}</td>
           <td class="crmf-is_admin crm-editable" data-type="boolean">{if $row.is_admin eq 1} {ts}Yes{/ts} {else} {ts}No{/ts} {/if}</td>
           <td class="nowrap crmf-weight">{$row.weight}</td>
           <td class="crmf-is_reserved">{if $row.is_reserved eq 1} {ts}Yes{/ts} {else} {ts}No{/ts} {/if}</td>
-          <td>{$row.action|replace:'xx':$row.id}</td>
+          <td>{if !empty($row.action)}{$row.action|replace:'xx':$row.id}{/if}</td>
```
Comment (eileenmcnaughton) [inline templates/CRM/Member/Page/MembershipStatus.tpl:51]: ok

File: templates/CRM/Group/Form/Edit.tpl [hunk 7]
```diff
@@ -36,7 +36,7 @@
     <tr class="crm-group-form-block-frontend-title">
       <td class="label">{$form.frontend_title.label} {if $action == 2}{include file='CRM/Core/I18n/Dialog.tpl' table='civicrm_group' field='frontend_title' id=$group.id}{/if}</td>
       <td>{$form.frontend_title.html|crmAddClass:huge}
-        {if $group.saved_search_id}&nbsp;({ts}Smart Group{/ts}){/if}
+        {if !empty($group.saved_search_id)}&nbsp;({ts}Smart Group{/ts}){/if}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Edit.tpl:39]: ok

File: templates/CRM/Group/Form/Edit.tpl [hunk 8]
```diff
@@ -69,14 +69,14 @@
       <td>{$form.is_active.html}</td>
     </tr>
 
-   {if $group.created_by}
+   {if !empty($group.created_by)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Edit.tpl:72]: ok

File: templates/CRM/Group/Form/Edit.tpl [hunk 9]
```diff
@@ -69,14 +69,14 @@
...
       <tr class="crm-group-form-block-created">
         <td class="label">{ts}Created By{/ts}</td>
         <td>{$group.created_by}</td>
       </tr>
     {/if}
 
-    {if $group.modified_by}
+    {if !empty($group.modified_by)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Edit.tpl:79]: ok

File: templates/CRM/Group/Form/Edit.tpl [hunk 10]
```diff
@@ -105,10 +105,10 @@
 
   {literal}
   <script type="text/javascript">
-    {/literal}{if $freezeMailingList}{literal}
+    {/literal}{if !empty($freezeMailingList)}{literal}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Edit.tpl:108]: ok

File: templates/CRM/Group/Form/Edit.tpl [hunk 11]
```diff
@@ -105,10 +105,10 @@
...
     cj('input[type=checkbox][name="group_type[{/literal}{$freezeMailingList}{literal}]"]').prop('disabled',true);
     {/literal}{/if}{literal}
-    {/literal}{if $hideMailingList}{literal}
+    {/literal}{if !empty($hideMailingList)}{literal}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Edit.tpl:111]: ok

File: templates/CRM/Group/Form/GroupsCommon.tpl [hunk 12]
```diff
@@ -8,9 +8,9 @@
  +--------------------------------------------------------------------+
 *}
 {*CRM-14190*}
-{if $parent_groups|@count > 0 or $form.parents.html}
+{if (isset($parent_groups) and $parent_groups|@count > 0) or !empty($form.parents.html)}
   <h3>{ts}Parent Groups{/ts} {help id="id-group-parent" file="CRM/Group/Page/Group.hlp"}</h3>
-  {if $parent_groups|@count > 0}
+  {if isset($parent_groups) and $parent_groups|@count > 0}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/GroupsCommon.tpl:13]: ok

File: templates/CRM/Group/Form/GroupsCommon.tpl [hunk 13]
```diff
@@ -8,9 +8,9 @@
  +--------------------------------------------------------------------+
 *}
 {*CRM-14190*}
-{if $parent_groups|@count > 0 or $form.parents.html}
+{if (isset($parent_groups) and $parent_groups|@count > 0) or !empty($form.parents.html)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/GroupsCommon.tpl:11]: ok

File: templates/CRM/Group/Form/GroupsCommon.tpl [hunk 14]
```diff
@@ -31,7 +31,7 @@
     </tr>
   </table>
 {/if}
-{if $form.organization_id}
+{if !empty($form.organization_id)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/GroupsCommon.tpl:34]: ok

File: templates/CRM/Group/Form/Search.tpl [hunk 15]
```diff
@@ -8,7 +8,7 @@
  +--------------------------------------------------------------------+
 *}
 <div class="crm-block crm-form-block crm-group-search-form-block">
-  <div class="crm-accordion-wrapper crm-search_builder-accordion {if $rows and !$showSearchForm}collapsed{/if}">
+  <div class="crm-accordion-wrapper crm-search_builder-accordion {if $rows and empty($showSearchForm)}collapsed{/if}">
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Search.tpl:11]: ok

File: templates/CRM/Group/Form/Search.tpl [hunk 16]
```diff
@@ -20,48 +20,58 @@
...
-            <td>
-              {$form.created_by.label}<br />
-              {$form.created_by.html}<br />
-              <span class="description font-italic">
-          {ts}Complete OR partial creator name.{/ts}
-      </span>
+                {ts}Complete OR partial group name.{/ts}
+              </span>
             </td>
+            {if !empty($form.created_by)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Search.tpl:26]: ok

File: templates/CRM/Group/Form/Search.tpl [hunk 17]
```diff
@@ -20,48 +20,58 @@
...
+              <td>
+                {$form.created_by.label}<br />
+                {$form.created_by.html}<br />
+                <span class="description font-italic">
+                  {ts}Complete OR partial creator name.{/ts}
+                </span>
+              </td>
+            {/if}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Search.tpl:34]: ok

File: templates/CRM/Group/Form/Search.tpl [hunk 18]
```diff
@@ -20,48 +20,58 @@
...
+              </span>
             </td>
           </tr>
           <tr>
-            <td>
-              {$form.saved_search.label} <br/>{$form.saved_search.html}
-            </td>
-            <td colspan="2">
-            </td>
+            {if !empty($form.group_type_search)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Search.tpl:44]: ok

File: templates/CRM/Group/Form/Search.tpl [hunk 19]
```diff
@@ -20,48 +20,58 @@
...
+              <td id="group_type-block">
+                {$form.group_type_search.label}<br />
+                {$form.group_type_search.html}<br />
+                <span class="description font-italic">
+                  {ts}Filter search by group type(s).{/ts}
+                </span>
+              </td>
+            {/if}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Search.tpl:52]: ok

File: templates/CRM/Group/Form/Search.tpl [hunk 20]
```diff
@@ -20,48 +20,58 @@
...
+            {if !empty($form.group_status)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Search.tpl:53]: ok

File: templates/CRM/Group/Form/Search.tpl [hunk 21]
```diff
@@ -20,48 +20,58 @@
...
+              <td>
+                {$form.group_status.label}<br />
+                {$form.group_status.html}
+              </td>
+            {/if}
+            {if !empty($form.component_mode)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Search.tpl:59]: ok

File: templates/CRM/Group/Form/Search.tpl [hunk 22]
```diff
@@ -20,48 +20,58 @@
...
+              <td>
+                {$form.component_mode.label}<br />
+                {$form.component_mode.html}
+              </td>
+            {/if}
           </tr>
+          {if !empty($form.saved_search)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Search.tpl:66]: ok

File: templates/CRM/Group/Form/Search.tpl [hunk 23]
```diff
@@ -81,7 +91,7 @@
       <th data-data="description" data-orderable="false" cell-class="crm-group-description crmf-description {$editableClass}" class='crm-group-description'>{ts}Description{/ts}</th>
       <th data-data="group_type" cell-class="crm-group-group_type" class='crm-group-group_type'>{ts}Group Type{/ts}</th>
       <th data-data="visibility" cell-class="crm-group-visibility crmf-visibility {$editableClass}" cell-data-type="select" class='crm-group-visibility'>{ts}Visibility{/ts}</th>
-      {if $showOrgInfo}
+      {if !empty($showOrgInfo)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Search.tpl:94]: ok

File: templates/CRM/Group/Form/Search.tpl [hunk 24]
```diff
@@ -125,7 +135,7 @@
           d.status = groupStatus,
           d.savedSearch = $('.crm-group-search-form-block select#saved_search').val(),
           d.component_mode = $(".crm-group-search-form-block select#component_mode").val(),
-          d.showOrgInfo = {/literal}"{$showOrgInfo}"{literal},
+          d.showOrgInfo = {/literal}{if isset($showOrgInfo)}"{$showOrgInfo}"{else}"0"{/if}{literal},
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Search.tpl:138]: ok - I guess I can see the reason for is to be 0 not empty

File: templates/CRM/Group/Form/Search.tpl [hunk 25]
```diff
@@ -174,7 +184,7 @@
     // show hide children
     var context = $('#crm-main-content-wrapper');
     $('table.crm-group-selector', context).on( 'click', 'span.show-children', function(){
-      var showOrgInfo = {/literal}"{$showOrgInfo}"{literal};
+      var showOrgInfo = {/literal}{if isset($showOrgInfo)}"{$showOrgInfo}"{else}"0"{/if}{literal};
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Search.tpl:187]: ok

File: templates/CRM/Mailing/Form/InsertTokens.tpl [hunk 26]
```diff
@@ -45,7 +45,7 @@ var isMailing    = false;
   {/literal}
 {/if}
 
-{if $templateSelected}
+{if !empty($templateSelected)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Mailing/Form/InsertTokens.tpl:48]: ok

File: templates/CRM/Member/Page/MembershipStatus.tpl [hunk 27]
```diff
@@ -38,17 +38,17 @@
             <th></th>
         </thead>
         {foreach from=$rows item=row}
-        <tr id="membership_status-{$row.id}" class="crm-entity {cycle values="odd-row,even-row"} {$row.class} {if NOT $row.is_active} disabled{/if} crmf">
+        <tr id="membership_status-{$row.id}" class="crm-entity {cycle values="odd-row,even-row"} {if !empty($row.class)}{$row.class}{/if} {if NOT $row.is_active} disabled{/if} crmf">
```
Comment (eileenmcnaughton) [inline templates/CRM/Member/Page/MembershipStatus.tpl:41]: ok

File: templates/CRM/Member/Page/MembershipStatus.tpl [hunk 28]
```diff
@@ -38,17 +38,17 @@
...
           <td class="crmf-label crm-editable" >{$row.label}</td>
-          <td class="nowrap crmf-start_event crm-editable" data-type="select" data-empty-option="{ts}- none -{/ts}">{$row.start_event}</td>
-          <td class="nowrap crmf-start_event_adjust_unit_interval">{$row.start_event_adjust_unit_interval}</td>
-          <td class="nowrap crmf-end_event crm-editable" data-type="select" data-empty-option="{ts}- none -{/ts}">{$row.end_event}</td>
-          <td class="nowrap crmf-end_event_adjust_interval">{$row.end_event_adjust_interval}</td>
+          <td class="nowrap crmf-start_event crm-editable" data-type="select" data-empty-option="{ts}- none -{/ts}">{if !empty($row.start_event)}{$row.start_event}{/if}</td>
```
Comment (eileenmcnaughton) [inline templates/CRM/Member/Page/MembershipStatus.tpl:43]: ok

File: templates/CRM/Member/Page/MembershipStatus.tpl [hunk 29]
```diff
@@ -38,17 +38,17 @@
...
+          <td class="nowrap crmf-start_event_adjust_unit_interval">{if !empty($start_event_adjust_unit_interval)}{$row.start_event_adjust_unit_interval}{/if}</td>
```
Comment (eileenmcnaughton) [inline templates/CRM/Member/Page/MembershipStatus.tpl:44]: this line and the line 2 down are checking a variable before presenting the key within the row array - is that right?

File: templates/CRM/Member/Page/MembershipStatus.tpl [hunk 30]
```diff
@@ -38,17 +38,17 @@
...
+          <td class="nowrap crmf-end_event crm-editable" data-type="select" data-empty-option="{ts}- none -{/ts}">{if !empty($row.end_event)}{$row.end_event}{/if}</td>
```
Comment (eileenmcnaughton) [inline templates/CRM/Member/Page/MembershipStatus.tpl:45]: ok

File: templates/CRM/Member/Page/MembershipStatus.tpl [hunk 31]
```diff
@@ -38,17 +38,17 @@
...
+          <td class="nowrap crmf-end_event_adjust_interval">{if !empty($end_event_adjust_unit_interval)}{$row.end_event_adjust_interval}{/if}</td>
```
Comment (eileenmcnaughton) [inline templates/CRM/Member/Page/MembershipStatus.tpl:46]: possible issue per above

**Comment by eileenmcnaughton** [comment]
OK - I think we got everything

**Pull request merged**